### ISSUE TITLE
Client Data Security Hardening

### ISSUE THREAD

**Issue by mgieselman** [mgieselman/LegalEagle#2]
## Overview

The application stores highly sensitive client data — SSNs, bank account numbers, tax returns, income, and other financial PII. This issue tracks the work needed to protect that data both at the access layer and by making it cryptographically useless if stolen.

---

## Current State (Gaps)

- All sensitive fields (SSNs, financial data) stored **plaintext** in `questionnaires.data` JSONB and `extractionResults.extractedData`
- Authentication is a dev stub — unauthenticated requests get admin access
- Documents stored unencrypted on the filesystem
- No audit log — no way to detect or investigate unauthorized access
- No database row-level security — only app-level `lawFirmId` filtering

---

### Auth (replace dev stub)
- **Azure AD B2C** _(recommended)_ — already on Azure; handles MFA, password reset, supports both staff (AAD) and client (B2C) flows. Free tier ~50k MAU.
- **Auth0** — easier DX, more expensive at scale
- **Clerk** — fast integration, good React SDK, less control
- **Self-hosted bcrypt + JWT** — full control, full responsibility (rate limiting, MFA, token rotation all manual)

### Database Row-Level Security (PostgreSQL)
Native RLS policies so the DB itself rejects cross-tenant reads, independent of app logic:
```sql
ALTER TABLE questionnaires ENABLE ROW LEVEL SECURITY;
CREATE POLICY tenant_isolation ON questionnaires
  USING (law_firm_id = current_setting('app.current_law_firm_id')::uuid);
```
Applies to production PostgreSQL only (not SQLite dev).

### Audit Logging
Log every read/write of sensitive data: `userId`, `lawFirmId`, `clientId`, `action`, `timestamp`, `ip`. Required in many jurisdictions for legal SaaS. Can be an `audit_log` table or shipped to Azure Monitor.

---

### Option A — Transparent Database Encryption (TDE) — Lowest effort
Encrypt at the infrastructure level. Azure Database for PostgreSQL already enables server-side encryption by default. Covers disk/backup theft but not a compromised app server or rogue DBA.

### Option B — Per-Client Field Encryption _(recommended)_ — Medium effort, high value
Generate a unique **Data Encryption Key (DEK)** per client, wrap it with a **Key Encryption Key (KEK)** stored in Azure Key Vault. Encrypt sensitive fields before they hit the DB.

```
Client created → generate random DEK (32 bytes)
→ wrap DEK with KEK in Azure Key Vault → store wrapped DEK in clients table
→ encrypt questionnaire.data with DEK → store ciphertext in DB

On read: unwrap DEK from Key Vault → decrypt → return to app
```

Fields to encrypt at minimum:
- `questionnaires.data` (SSN, financial data, bank accounts)
- `extractionResults.extractedData`
- `clients.phone`, `clients.email`

**Protects against:** DB dump, stolen backups, insider threat at DB layer. A thief gets ciphertext — without Key Vault access it is useless.

### Option C — Bring Your Own Key (BYOK) — High effort, enterprise feature
Each law firm controls their own KEK in their own Azure Key Vault. LegalEagle never holds the KEK. Protects against even a full platform breach. Key loss = data loss; complex to operate. Late-stage enterprise feature.

### Option D — Column-Level Encryption via pgcrypto — Simple, limited
```sql
UPDATE questionnaires SET data = pgp_sym_encrypt(data::text, $key);
```
Moves encryption into the DB process. Simpler operationally but the key must be accessible to Postgres, which limits the threat model vs. Option B.

---

## Recommended Roadmap

| Priority | Item |
|---|---|
| **P0 — Before any real data** | Replace dev auth with Azure AD B2C |
| **P0** | Enable PostgreSQL RLS for tenant isolation |
| **P0** | Add audit logging table |
| **P1 — Before pilot launch** | Per-client DEK encryption (Option B) for `questionnaires.data` and `extractionResults.extractedData` using Azure Key Vault |
| **P1** | Documents encrypted at rest in Azure Blob Storage (built-in, just needs enabling) |
| **P1** | Rate limiting + session timeouts |
| **P2 — Enterprise** | BYOK for law firm tenants |
| **P2** | Penetration test + SOC 2 Type II |
